### PR TITLE
Add support for Scrollable single element screenshot

### PR DESCRIPTION
**Pull request by chinmay-browserstack**
Added new option scope-options in which user can pass scroll: true/false to take scrollable single element screenshot properly

### CODE REVIEW

File: packages/core/src/config.js [hunk 1]
```diff
@@ -59,6 +59,15 @@ export const configSchema = {
       scope: {
         type: 'string'
       },
+      scopeOptions: {
+        type: 'object',
```
Comment (ninadbstack) [inline packages/core/src/config.js:63]: For all objects we should define valid keys for the same, so that users can understand if they have typos

Comment (chinmay-browserstack) [reply packages/core/src/config.js:63]: This are valid keys for given option https://github.com/percy/cli/pull/1458/files#diff-e7c1269ddb64f1de14c2c79cac204c16cfc3f8d8535512d218f8e4ad8efc63b6R66-R68. anything else need to be added?

File: packages/core/src/config.js [hunk 2]
```diff
@@ -59,6 +59,15 @@ export const configSchema = {
...
+        additionalProperties: false,
+        properties: {
+          scroll: {
+            type: 'boolean'
```
Comment (itsjwala) [inline packages/core/src/config.js:67]: can you update typescript definition index.ts?

File: packages/core/src/config.js [hunk 3]
```diff
@@ -59,6 +59,15 @@ export const configSchema = {
...
+          }
+        }
+      },
```
Comment (itsjwala) [inline packages/core/src/config.js:70]: add specs for same, with and without this config.

File: packages/core/src/discovery.js [hunk 4]
```diff
@@ -42,6 +42,7 @@ function debugSnapshotOptions(snapshot) {
   debugProp(snapshot, 'deviceScaleFactor');
   debugProp(snapshot, 'waitForTimeout');
   debugProp(snapshot, 'waitForSelector');
+  debugProp(snapshot, 'scopeOptions.scroll');
```
Comment (itsjwala) [inline packages/core/src/discovery.js:45]: update specs

File: packages/client/src/client.js [hunk 5]
```diff
@@ -329,6 +330,7 @@ export class PercyClient {
           name: name || null,
           widths: widths || null,
           scope: scope || null,
+          'scope-options': scopeOptions || {},
```
Comment (itsjwala) [inline packages/client/src/client.js:333]: add specs when value given

File: packages/core/src/config.js [hunk 6]
```diff
@@ -244,6 +253,13 @@ export const snapshotSchema = {
         domTransformation: { $ref: '/config/snapshot#/properties/domTransformation' },
         enableLayout: { $ref: '/config/snapshot#/properties/enableLayout' },
         reshuffleInvalidTags: { $ref: '/config/snapshot#/properties/reshuffleInvalidTags' },
+        scopeOptions: {
```
Comment (itsjwala) [inline packages/core/src/config.js:256]: it would have same definition as L62 ( make it similar to above lines)

Comment (chinmay-browserstack) [reply packages/core/src/config.js:256]: It is same, and this is how it is defined for other properties

**Review by itsjwala — APPROVED** [review]
we'll check test failure separately as its not related to this PR, merging to unblock

**Pull request merged**